src/test/java/picard/fingerprint/CheckFingerprintTest.java

```diff
@@ -272,21 +288,45 @@ void testCheckFingerprintNoRg(final File file, final File genotypes, final int e
         tester(true, file, genotypes, expectedRetVal);
     }
 
-    private File tester(boolean ignoreRG, final File file, final File genotypes, final int expectedRetVal) throws IOException {
+    private List<String> buildArgs(boolean ignoreRG, final File file, final File genotypes) {
         final List<String> args = new ArrayList<>();
-        final File outputSummary = File.createTempFile("fingerprint", "summary_metrics");
-        outputSummary.deleteOnExit();
-        final File outputDetail = File.createTempFile("fingerprint", "detail_metrics");
-        outputSummary.deleteOnExit();
 
         args.add("INPUT=" + file.getAbsolutePath());
         args.add("G=" + genotypes.getAbsolutePath());
         if (ignoreRG) {
             args.add("IGNORE_RG=true");
         }
         args.add("H=" + HAPLOTYPE_MAP.getAbsolutePath());
+
+        return args;
+    }
+
+    private File tester(boolean ignoreRG, final File file, final File genotypes, final int expectedRetVal) throws IOException {
+        final List<String> args = buildArgs(ignoreRG, file, genotypes);
+        final File outputSummary = File.createTempFile("fingerprint", "summary_metrics");
+        outputSummary.deleteOnExit();
+        final File outputDetail = File.createTempFile("fingerprint", "detail_metrics");
+        outputDetail.deleteOnExit();
+        args.add("SUMMARY_OUTPUT=" + outputSummary.getAbsolutePath());
+        args.add("DETAIL_OUTPUT=" + outputDetail.getAbsolutePath());
+
+        Assert.assertEquals(runPicardCommandLine(args), expectedRetVal);
+
+        Assert.assertTrue(outputSummary.exists(), "Expected output file " + outputSummary.getAbsolutePath() + " to exist.");
+        Assert.assertTrue(outputDetail.exists(), "Expected output file " + outputDetail.getAbsolutePath() + " to exist.");
+
+        return outputSummary;
+    }
+
+    private File testWithMinBaseQ(boolean ignoreRG, final File file, final File genotypes, final int expectedRetVal, final int minBaseQ) throws IOException {
```

Comment:
since this is basically identical to `tester` what about adding an array input of `additionalArgs` to tests that gets appended to the arguments, and just using that to reduce the duplicate code